### PR TITLE
Rework filtering:

### PR DESCRIPTION
**Pull request by velkuns**
- Change priority & status into array of filter.
 - Better filtering (with dates)
 - Now ordering by priority & availability first.
 - Update README
 - Update sample

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,32 @@ You must have a database with given table to store messages (the messages queue)
...
+
+### Filtering
+You may filtering message you want to get. Possible filters are:
+ - topic:  specify which topic you want (default: null). You can use `*` character in topic filter (like: publish.*)
+ - statuses: Get messages with given status(es) (default get pending messages)
+ - priorities: Get messages with only given priority(ies) (default: no filtering)
+ - entity id: Filter only on the given entity id (default: no filtering)
+
+### Filtering on date
+By default, the system get only the message available and non-expired.
```
Comment (pasdeloup) [inline README.md]: By default, the system only gets the available and non-expired messages.

File: README.md [hunk 2]
```diff
@@ -13,6 +13,32 @@ You must have a database with given table to store messages (the messages queue)
...
+Functions getMessage() & getMessages get messages in following order:
+ - priority ascendant
+ - date availability ascendant
+ - date create ascendant
+
+So, the messages with high priority, and with prior date availability (and with prior creation date)
+will be get in priority.
+
+### Filtering
+You may filtering message you want to get. Possible filters are:
```
Comment (pasdeloup) [inline README.md]: You may filter messages ...

File: README.md [hunk 3]
```diff
@@ -13,6 +13,32 @@ You must have a database with given table to store messages (the messages queue)
...
+ - statuses: Get messages with given status(es) (default get pending messages)
+ - priorities: Get messages with only given priority(ies) (default: no filtering)
+ - entity id: Filter only on the given entity id (default: no filtering)
+
+### Filtering on date
+By default, the system get only the message available and non-expired.
+You can change filter value for available date & expiration date. You also can change the current date.
+
+### Limit result
+By default, getMessage() get only one message.
```
Comment (pasdeloup) [inline README.md]: By default, getMessage() only gets one message.

File: README.md [hunk 4]
```diff
@@ -13,6 +13,32 @@ You must have a database with given table to store messages (the messages queue)
...
+You may get multiple message at the same time by using getMessages(), but you need to set how much message you want (default is 1)
```
Comment (pasdeloup) [inline README.md]: messages

**Comment by velkuns** [comment]
**QA+1**

**Review by pasdeloup — APPROVED** [review]

**Pull request merged**